lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1161,15 +1175,43 @@ sub create_result_dir {
     return $dir;
 }
 
+my %JOB_MODULE_STATISTICS_COLUMN_BY_JOB_MODULE_RESULT = (
+    OpenQA::Jobs::Constants::PASSED     => 'passed_module_count',
+    OpenQA::Jobs::Constants::SOFTFAILED => 'softfailed_module_count',
+    OpenQA::Jobs::Constants::FAILED     => 'failed_module_count',
+    OpenQA::Jobs::Constants::NONE       => 'skipped_module_count',
+    OpenQA::Jobs::Constants::SKIPPED    => 'externally_skipped_module_count',
+);
+
+sub _get_job_module_statistics_column_by_job_module_result {
+    my ($job_module_result) = @_;
+    return undef unless defined $job_module_result;
+    return $JOB_MODULE_STATISTICS_COLUMN_BY_JOB_MODULE_RESULT{$job_module_result};
+}
+
 sub update_module {
-    my ($self, $name, $result) = @_;
+    my ($self, $name, $raw_result) = @_;
 
+    # find the module
+    # FIXME: There is no unique constraint for just the job_id and the name so search() should be used instead
+    #        of find().
```

Comment:
Is there a ticket for it?

Reply:
I'll create a PR (it will be based on this one to prevent a conflict). I don't want to make it part of this PR so it can be merged as quick as possible to resolve the urgent ticket.